### PR TITLE
[Angular] Rename ConfigService to ApplicationConfigService

### LINKED ISSUE #14109: [Angular] ConfigService should be renamed
##### **Overview of the issue**
The only function of config.service.ts is either to set the endpoint prefix or get the endpoint. I propose renaming the class to `EndpointService` or `EndpointConfigService` to make more sense of its functions.

##### **Motivation for or Use Case**
Users having already an (or creating a new) entity called `Config` will get the following error:
`Import declaration conflicts with local declaration of 'ConfigService'`

Which means the class name is the same as a class we're importing. Keeping the current name will result in errors when upgrading to users having said entity.

Another motivation would be to have a class with a name that tells what does the class do as "ConfigService" is too generic.

##### **Reproduce the error**

While using the main branch
* create a new project with Angular as frontend 
* add an entity called Config
* Open entity's ConfigService.

##### **Related issues**

None that I know of.

##### **Suggest a Fix**

Rename ConfigService class to either `EndpointService` or `EndpointConfigService` (or any other name that makes sense) so users that already have an entity called Config will have no issue migrating.

##### **JHipster Version(s)**
7.0.0-beta.1

##### **JHipster configuration**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.amatosg"
    },
    "jhipsterVersion": "7.0.0-beta.1",
    "applicationType": "monolith",
    "baseName": "ConfTestProject",
    "packageName": "com.amatosg",
    "packageFolder": "com/amatosg",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "caffeine",
    "enableHibernateCache": true,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "postgresql",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "clientTheme": "spacelab",
    "clientThemeVariant": "primary",
    "testFrameworks": [],
    "jhiPrefix": "fa",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": false,
    "blueprints": [],
    "creationTimestamp": 1580194041932,
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "pages": [],
    "withAdminUi": true,
    "nativeLanguage": "en",
    "languages": ["en", "fr"],
    "reactive": false,
    "lastLiquibaseTimestamp": 1614565273000,
    "entities": ["Config"]
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Config {
  description String required
}
dto Config with mapstruct
paginate Config with pagination
service Config with serviceImpl
filter Config

</pre>
</details>

##### **Browsers and Operating System**
Not related.

- [x] Checking this box is mandatory (this is just to show you read everything)

Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #14109 

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/cleanup.js [hunk 1]
```diff
@@ -265,6 +265,8 @@ function cleanupOldFiles(generator) {
       generator.removeFile(`${ANGULAR_DIR}core/user/user.model.ts`);
       generator.removeFile(`${ANGULAR_DIR}core/user/user.service.ts`);
       generator.removeFile(`${ANGULAR_DIR}core/user/user.service.spec.ts`);
+      generator.removeFile(`${ANGULAR_DIR}core/config/application-config.service.ts`);
+      generator.removeFile(`${ANGULAR_DIR}core/config/application-config.service.spec.ts`);
```
Comment (mshima) [inline generators/cleanup.js:269]:
```suggestion
      generator.removeFile(`${ANGULAR_DIR}core/config/config.service.ts`);
      generator.removeFile(`${ANGULAR_DIR}core/config/config.service.spec.ts`);
```
And create generator.isJhipsterVersionLessThan('7.0.0-beta.2')

**Review by mshima — APPROVED** [review]
LGTM.
Just adjust the file removal.

File: generators/cleanup.js (hunk 1 above)
Comment (qmonmert) [reply generators/cleanup.js]: done

File: generators/cleanup.js [hunk 2]
```diff
@@ -265,6 +265,8 @@ function cleanupOldFiles(generator) {
       generator.removeFile(`${ANGULAR_DIR}core/user/user.model.ts`);
       generator.removeFile(`${ANGULAR_DIR}core/user/user.service.ts`);
       generator.removeFile(`${ANGULAR_DIR}core/user/user.service.spec.ts`);
+      generator.removeFile(`${ANGULAR_DIR}core/config/config.service.ts`);
+      generator.removeFile(`${ANGULAR_DIR}core/config/config.service.spec.ts`);
```
Comment (mshima) [inline generators/cleanup.js:269]:
```suggestion
```
Can be dropped now.

Comment (qmonmert) [reply generators/cleanup.js]: sorry, yes of course

Comment (mshima) [reply generators/cleanup.js]: no problem 😄.

**Comment by qmonmert** [comment]
hello @mshima do you think that we can merge? 😇

**Comment by mshima** [comment]
I'm ok to merge, unless @pascalgrimaud wants to use jhi_prefix.

**Comment by pascalgrimaud** [comment]
I'm fine with this, it was just an idea to use jhi prefix

**Pull request merged**